java-symbolic-execution/java-symbolic-execution-plugin/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -474,11 +474,21 @@ private void closeResource(@Nullable final SymbolicValue target) {
     @Override
     public void visitIdentifier(IdentifierTree tree) {
       // close resource as soon as it is encountered in the resource declaration
-      if (isWithinTryHeader(tree)) {
+      // or if it is annotated with @lombok.Cleanup
+      if (isWithinTryHeader(tree) || isAnnotatedLombokCleanup(tree)) {
         Symbol symbol = tree.symbol();
         closeResource(programState.getValue(symbol));
       }
     }
+
+    private static boolean isAnnotatedLombokCleanup(IdentifierTree tree) {
+      return tree
+        .symbol()
+        .metadata()
+        .annotations()
+        .stream()
+        .anyMatch(annotation -> "lombok.Cleanup".equals(annotation.symbol().type().fullyQualifiedName()));
```

Comment:
Let's try to support the case where we do not have semantics but the annotation looks like `Cleanup` or `lombok.Cleanup`